akka-projection-testkit/src/main/scala/akka/projection/testkit/ProjectionTestKit.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.testkit
+
+import akka.Done
+import akka.actor.testkit.typed.TestKitSettings
+import akka.actor.testkit.typed.scaladsl.{ ActorTestKit, _ }
+import akka.annotation.ApiMayChange
+import akka.projection.{ Projection, RunningProjection }
+import akka.stream.scaladsl.Sink
+import akka.stream.testkit.TestSubscriber
+import akka.stream.testkit.scaladsl.TestSink
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Future, Promise }
+import akka.actor.typed.scaladsl.adapter._
+
+@ApiMayChange
+object ProjectionTestKit {
+  def apply(testKit: ActorTestKit): ProjectionTestKit =
+    new ProjectionTestKit(testKit)
+}
+
+@ApiMayChange
+final class ProjectionTestKit private[akka] (testKit: ActorTestKit) {
+
+  private implicit val system = testKit.system
+  private implicit val settings: TestKitSettings = TestKitSettings(system)
+
+  def run(projection: Projection)(testFunc: => Unit): Future[Done] =
+    runInternal(projection, testFunc, settings.SingleExpectDefaultTimeout.dilated, 100.millis)
+
+  def run(projection: Projection, max: FiniteDuration)(testFunc: => Unit): Future[Done] = {
+    runInternal(projection, testFunc, max, 100.millis)
+  }
+
+  def run(projection: Projection, max: FiniteDuration, interval: FiniteDuration = 100.millis)(
+      testFunc: => Unit): Future[Done] = {
+    runInternal(projection, testFunc, max, 100.millis)
+  }
+
+  private def runInternal(
+      projection: Projection,
+      testFunc: => Unit,
+      max: FiniteDuration,
+      interval: FiniteDuration = 100.millis): Future[Done] = {
+
+    val probe = testKit.createTestProbe[Nothing]("internal-projection-testkit-probe")
+
+    val promiseToStop = Promise[Done]
+    val runningProjection: RunningProjection = projection.run()
+
+    try {
+      probe.awaitAssert(testFunc, max, interval)
+    } finally {
+      promiseToStop.completeWith(runningProjection.stop())
+    }
+    promiseToStop.future
+  }
+
+  def runWithTestSink[T](proj: Projection): TestSubscriber.Probe[Done] = {
+    val sinkProbe = TestSink.probe[Done](system.toClassic)
+    proj.mappedSource.runWith(sinkProbe)
+  }
```

Comment:
In addition to the `run` + `assertFunction` variants, we have this extra `runWithTestSink` that will return a probe that we can use to control the elements.